lib/virtio_mmio/virtio_mmio.c

```diff
@@ -302,6 +371,26 @@ int virtio_mmio_device_init(struct virtio_mmio_device *vmdev, uintptr_t virt_mem
 	vdev->id.device = devid;
 	vdev->id.vendor = vendor;
 
+#if defined(HVL_VIRTIO)
+	/* Check shared memory information.
+	 * This is just a flag for enabling hypervisorless mode for now.
+	 */
+	shm_base = virtio_mmio_read32(vdev, VIRTIO_MMIO_SHM_BASE_HIGH);
```

Comment:
what contain this shared memory ? the bounce buffers?

Reply:
Virtio device drivers (aka virtio front-ends) have the liberty to allocate memory anywhere in their memory space. In order to be able to reuse these drivers without changes, a bounce buffer indirection layer is used. However, if the virtio front-end uses data buffers allocated from the shared memory region, "bouncing" is disabled.